Reduce the matrix combination on Fedora

tuliom opened the pull request:
Keep running the complete set of tests on Rawhide. Keep running the tests on Fedora N-1, except for s390x. Reduce the list of architectures for Fedora N-2 to aarch64 and x86_64. Which also means that i386 and ppc64le won't be tested on Fedora N-2.

nikic commented:
> 1. We cannot assume the latest Fedora version is `rawhide`.

I don't think we need to assume this. The rule here is that for rawhide we want to build all chroots. If there is no rawhide, then that rule just doesn't apply. There is no need to treat the highest numbered fedora version as if it were rawhide instead. We can continue treating it like the highest branched release.

Similar for the numbered releases. If there are gaps in the releases (like we only get fedora-41 and fedora-43, but not fedora-42), then there is no need to go out of the way in order to treat fedora-41 as if it were actually fedora-42, rather than just not building it at all, because there are no chroots for branched-1.

I feel like unnecessary test coverage that has been added at some point is forcing this into weird directions now. It's good to make sure that we don't crash if we get unexpected input, but we shouldn't write more complicated code just to satisfy those tests.

tuliom commented:
@nikic I've changed the behavior to return rawhide, N and N-1, if available.
I also modified the return type in order to simplify the caller function (tuple instead of list).

nikic reviewed (approved):
LGTM. This is still not quite what I had in mind, but I don't want to nitpick it too much.

tuliom commented on snapshot_manager/snapshot_manager/util.py:
```diff
@@ -591,52 +666,34 @@ def sanitize_chroots(chroots: list[str]) -> list[str]:
+        for chroot in chroots
         if chroot_name(chroot) != "fedora"
-        or chroot_version(chroot) in fedora_versions_to_keep
+        or chroot_version(chroot) == rawhide_version
```
@nikic Agreed. I completely forgot about this in the previous iteration.

I implemented this now and I ended up changing the return of the function because it didn't make anymore, IMO.
